sweyer_plugin/lib/sweyer_plugin_method_channel.dart

```diff
@@ -152,8 +132,11 @@ class MethodChannelSweyerPlugin extends SweyerPluginPlatform {
           'name': name,
         },
       );
-    } on PlatformException catch (ex) {
-      throw SweyerMethodChannelException._throw(ex, const [SweyerMethodChannelException.playlistNotExists]);
+    } on PlatformException catch (error) {
+      if (error.code == SweyerMethodChannelExceptionCode.playlistNotExists.value) {
+        throw PlaylistNotExistException(playlistId, cause: error);
```

Comment:
What is the differentiation between "common" and not common? Why `_convertCommonExceptions` doesn't just handle all of them?

Why is this method not a part of the platform interface?

Reply:
> What is the differentiation between "common" and not common?

Well, the "common" exceptions don't have a parameter like the `playlistId`. Please suggest a better name if you have one, I'm also not too thrilled with the current one.

>  Why _convertCommonExceptions doesn't just handle all of them?

Do you think it would be better to pass the `playlistId` as an optional parameter? This might get ugly in the future, if we add other errors with parameters.

> Why is this method not a part of the platform interface?

Because this converts `PlatformException`s, which are specific to method channels. A mock implementation of the `SweyerPluginPlatform` interface or any other theoretical implementation that does not use method channels does not need this method.

Reply:
Ah, I see now, thanks